### PR TITLE
Making binding name mandatory (remove defaulting rules)

### PR DESCRIPTION
**Pull request by mathewc**
We've decided to simplify things by making input binding names required, removing the defaulting rules we had in place (e.g. default to "input"). Ling's recent PowerShell work pointed out a problem - in the new PowerShell host, "input" is not valid since it's a reserved name in PowerShell. So rather than introduce language specific defaults, we'll just require a name from the user.

This is a breaking change, and will be released with other breaking changes (the new PowerShell model is breaking) in the next release (we'll rev our major version). The impact from these naming changes should be very small however, since all our templates already explicitly set a name.

### CODE REVIEW

File: src/WebJobs.Script/Description/FunctionDescriptorProvider.cs [hunk 1]
```diff
@@ -406,24 +416,5 @@ protected ParameterDescriptor ParseApiHubFileTrigger(ApiHubBindingMetadata trigg
             };
             return new ParameterDescriptor(parameterName, triggerParameterType, attributes);
         }
-
-        private void DefaultTriggerBindingParameterName(BindingMetadata triggerBindingMetadata)
```
Comment (mathewc) [inline src/WebJobs.Script/Description/FunctionDescriptorProvider.cs]: No more name defaulting :)

File: src/WebJobs.Script/Description/DotNet/DotNetFunctionInvoker.cs [hunk 2]
```diff
@@ -82,7 +80,7 @@ private static string GetTriggerInputName(FunctionMetadata functionMetadata, str
                 triggerName = triggerBinding.Name;
             }
 
-            return triggerName ?? defaultInputParameterName;
+            return triggerName;
```
Comment (fabiocav) [inline src/WebJobs.Script/Description/DotNet/DotNetFunctionInvoker.cs]: Should we ever return `null` from this? Should we be failing if we can't resolve the name?

By the way, now we should be able to use `triggerBinding?.Name` or return straight from `functionMetadata.Bindings.FirstOrDefault(b => b.IsTrigger)?.Name`

Comment (mathewc) [inline src/WebJobs.Script/Description/DotNet/DotNetFunctionInvoker.cs]: I've added validation upstream (see FunctionDescriptorProvider.ValidateFunction), as well as new test suite FunctionDescriptionProviderTests

**Comment by fabiocav** [comment]
Looks good! 👍

**Pull request merged**